src/util/impl_const_id.rs

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! A simple const generics substitute.
```

Comment:
I think min const generics were stabilized at some point.

@npmccallum @haraldh Do you have a preference for how aggressive we should be with how old of a Rust toolchain this crate supports?

Reply:
We support the use of min const generics (we need it for other things).

I don't think we should bother supporting old Rust. We can even support nightly features (where clearly useful and where there is a path to stabilization). RH is currently rebasing the toolchain regularly, so I don't see much problem with this.